src/rgw/rgw_bucket_layout.h

```diff
@@ -72,6 +72,46 @@ void encode(const bucket_index_layout_generation& l, bufferlist& bl, uint64_t f=
 void decode(bucket_index_layout_generation& l, bufferlist::const_iterator& bl);
 
 
+enum class BucketLogType : uint8_t {
+  // colocated with bucket index, so the log layout matches the index layout
+  InIndex,
+};
+
+inline std::ostream& operator<<(std::ostream& out, const BucketLogType &log_type)
+{
+  switch (log_type) {
+    case BucketLogType::InIndex:
+      return out << "InIndex";
+    default:
+      return out << "Unknown";
+  }
+}
+
+struct bucket_log_layout {
+  BucketLogType type = BucketLogType::InIndex;
+
+  bucket_index_normal_layout in_index;
```

Comment:
this `InIndex` log layout will need to track the index layout's generation too - we can't assume that it will match `bucket_log_layout_generation::gen` after the changes to `bucket sync enable` in https://github.com/ceph/ceph/pull/38010